core/sql/optimizer/NATable.cpp

```diff
@@ -5008,541 +5009,546 @@ NABoolean createNAFileSets(TrafDesc * table_desc       /*IN*/,
-				  this             /*IN*/,
-				  colArray_        /*IN*/,
-				  indexes_         /*OUT*/,
-				  vertParts_       /*OUT*/,
-				  clusteringIndex_ /*OUT*/,
-				  tableIdList_     /*OUT*/,
-				  heap_,
-				  bindWA,
-				  newColumns_,     /*OUT*/
-				  maxIndexLevelsPtr)) {
-		  return; // colcount_ == 0 indicates an error
-	  }
+       //
+       // The updatable flag is false for an MP view only if it is NOT a
+       // protection view. Therefore updatable == FALSE iff it is a
+       // shorthand view. See ReadTableDef.cpp, l. 3379.
+       //
 
-	  // Add constraint info.
-	  //
-	  // This call to createConstraintInfo, calls the parser on
-	  // the constraint name
-	  //
+       viewFileName_ = NULL;
+       CMPASSERT(view_desc->viewDesc()->viewfilename);
+       UInt32 viewFileNameLength = str_len(view_desc->viewDesc()->viewfilename) + 1;
+       viewFileName_ = new (heap_) char[viewFileNameLength];
+       memcpy(viewFileName_, view_desc->viewDesc()->viewfilename,
+              viewFileNameLength);
+     }
+   else
+     {
+       //keep track of memory used by NAFileSets
+       Lng32 preCreateNAFileSetsMemSize = heap_->getAllocSize();
 
-	  NABoolean  errorOccurred =
-		  createConstraintInfo(table_desc        /*IN*/,
-				  getTableName()    /*IN*/,
-				  getNAColumnArray()/*IN (some columns updated)*/,
-				  checkConstraints_ /*OUT*/,
-				  uniqueConstraints_/*OUT*/,
-				  refConstraints_   /*OUT*/,
-				  heap_,
-				  bindWA);
-
-	  if (errorOccurred) {
-		  // return before setting colcount_, indicating that there
-		  // was an error in constructing this NATable.
-		  //
-		  return;
-	  }
+       //
+       // Process indexes and vertical partitions for this table.
+       //
+       if (createNAFileSets(table_desc       /*IN*/,
+                            this             /*IN*/,
+                            colArray_        /*IN*/,
+                            indexes_         /*OUT*/,
+                            vertParts_       /*OUT*/,
+                            clusteringIndex_ /*OUT*/,
+                            tableIdList_     /*OUT*/,
+                            heap_,
+                            bindWA,
+                            newColumns_,     /*OUT*/
+                            maxIndexLevelsPtr)) {
+         return; // colcount_ == 0 indicates an error
+       }
 
-	  //
-	  // FetchHistograms call used to be here -- moved to getStatistics().
-	  //
-  }
+       // Add constraint info.
+       //
+       // This call to createConstraintInfo, calls the parser on
+       // the constraint name
+       //
 
-  // change partFunc for base table if PARTITION clause has been used
-  // to limit the number of partitions that will be accessed.
-  if ((qualifiedName_.isPartitionNameSpecified()) ||
-		  (qualifiedName_.isPartitionRangeSpecified())) {
-	  if (filterUnusedPartitions(corrName.getPartnClause())) {
-		  return ;
-	  }
-  }
+       NABoolean  errorOccurred =
```

Comment:
No, we dont support constraints on hive external tables. An error will be returned if unique, foreign, or check constraint is specified while creating a hive external table. 

On the other question of IMPLICIT keyword, so that can only be used when the create stmt
is issued internally and the internal flag is set. If used externally, a syntax error will be returned.
From functionality pov, there is no difference between an explicit or implicit external table.

In showddl output, the implicit keyword shows up as an informational comment:
  create /*implicit*/ external table...
This is similar to how we also show traf internally created system indexes.